File: 2º Ano/2º Semestre/CP/projeto/testing.c
#include <stdio.h>
#include <string.h>

int inv(int x, int n){
	int sum=0;
	for(int i=0; i<=n; i++){
		sum+=(1-x)^i;
	}
	return sum;
}

int main(){
	printf("inv(1.5,3) = %f\n",inv(1.5,3));
	return 0;
}

File: 2º Ano/2º Semestre/CP/projeto/testing.hs
inv :: Float -> Int -> Float
inv 0 _ = 1
inv _ 0 = 1
inv x n = (1-x)^n + inv x (n-1)

bla :: Float -> Int -> Float
bla 0 _ = 1
bla _ 0 = 1
bla x n = (1-x)*(bla x (n-1))

mrd :: Float -> Float 
mrd 0 = 1
mrd x = 1-(mrd (x-1))